soccer/gameplay/tactics/advance_zone_midfielder.py

```diff
@@ -0,0 +1,182 @@
+        self.add_transition(behavior.Behavior.State.start,
+                            AdvanceZoneMidfielder.State.passSet, lambda: True,
+                            "Immediately")
+        self.add_transition(
+            AdvanceZoneMidfielder.State.passSet,
+            AdvanceZoneMidfielder.State.hold, lambda: self.kick,
+            "When a kick play begins")
+        self.add_transition(
+            AdvanceZoneMidfielder.State.hold,
+            AdvanceZoneMidfielder.State.passSet, lambda: not self.kick,
+            "When not in a kick play")
+
+    def execute_passSet(self):
+        #gets the best position to travel to for ball reception
+        points = [robocup.Point(0, 0), robocup.Point(0, 0)]
+        points[0] = self.passing_point
+
+        # sets the second point
+        points[
+            1], value2 = evaluation.passing_positioning.eval_best_receive_point(
+                self.passing_point,
+                main.our_robots(), AdvanceZoneMidfielder.FIELD_POS_WEIGHTS,
+                AdvanceZoneMidfielder.NELDER_MEAD_ARGS,
+                AdvanceZoneMidfielder.PASSING_WEIGHTS)
+
+        # check for futile position
+        if self.in_shot_triangle(points):
+            points[1] = self.remove_obstruction(points)
+
+        # moves the robots and assigns information
+        for i in range(len(points)):
+            if (self.moves[i] is None):
+                self.moves[i] = skills.move.Move(points[i])
+                self.add_subbehavior(
+                    self.moves[i],
+                    self.names[i],
+                    required=False,
+                    priority=self.priorities[i])
+            else:
+                self.moves[i].pos = points[i]
+
+    def execute_hold(self):
+        # old methodology for simple zone midfielder (check comments on that method)
+        y_temp_hold = 0.8 * self.passing_point.y
+
+        x_temp_hold = constants.Field.Width / 3
+
+        self.hold_point = [
+            robocup.Point(x_temp_hold, y_temp_hold),
+            robocup.Point(-x_temp_hold, y_temp_hold)
+        ]
+
+        for i in range(len(self.hold_point)):
+            if (self.moves[i] is None):
+                self.moves[i] = skills.move.Move(self.hold_point[i])
+                self.add_subbehavior(
+                    self.moves[i], self.names[i], required=False, priority=1)
+            else:
+                self.moves[i].pos = self.hold_point[i]
+
+    # gets passing point from adaptive formation
+    @property
+    def passing_point(self):
+        return self._passing_point
+
+    @passing_point.setter
+    def passing_point(self, value):
+        self._passing_point = value
+
+    # gets when we are kicking from adaptive formation
+    @property
+    def kick(self):
+        return self._kick
+
+    @kick.setter
+    def kick(self, value):
+        self._kick = value
+
+    def on_exit_passSet(self):
+        self.remove_all_subbehaviors()
+
+    def on_exit_hold(self):
```

Comment:
This might be super trivial, but it looks better when reading code to have all the setter and getter's below almost all the other code. I also personally like having the functions that we use be directly below the functions they are being called. So if function A calls function B which calls function C. The order would be A, B ,C in the file.

Reply:
Moved it to the end